### ISSUE TITLE
Newly created pages appear as folders in the sidebar

### ISSUE THREAD

**Issue by alhambrav** [craftercms/craftercms#6589]
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] The issue is in the develop branch
- [X] The issue is in the latest released 4.1.x
- [ ] The issue is in the latest released 4.0.x
- [ ] The issue is in the latest released 3.1.x

### Describe the issue

New articles created via copy-pasting, duplicating, or new content appear as folders in the sidebar. Once you refresh, the correct icons appear. 

![Screenshot 2024-02-27 at 9 20 39 AM](https://github.com/craftercms/craftercms/assets/25483966/33d4a33e-0729-4a20-93fc-d3865fe95cd4)

CrafterCMS running on MacOS 14.2.1 using a Chrome browser

### Steps to reproduce

Steps:

1. Create a project using the Website Editorial bp
2. Open the Sidebar and navigate to say the folder `articles/2020/12`
3. Duplicate the article `Top Books for Young Women`
4. Notice that the new duplicate article has the folder icon in the Sidebar
![Screenshot 2024-02-28 at 9 42 41 AM](https://github.com/craftercms/craftercms/assets/25483966/9e95cfe1-6a9f-4d3a-8469-a1145a4a3843)
5. You can also try creating a new article or copy-pasting an article and it shows a folder icon in the Sidebar until you refresh then it shows the correct icon

### Screenshots and/or videos

https://www.loom.com/share/791c8571c339476daec8b6b547ad3191

**Comment by rart** [comment]
For context, the issue doesn't happen every time and the operation(s) executes correctly on all fronts. The issue is a **race condition**, pertaining the way the socket sends page creation events. The socket sends 2 consecutive socket events, one for the creation of the folder and one for the creation of the index.

Related to https://github.com/craftercms/craftercms/issues/6478, as the UI dedupe might help but we might also want to evaluate if the back should send the folder creation event at all when the operation is a page creation.

Event 1:
```js
{
  timestamp: 1709153527537,
  user: { ... },
  siteId: 'editorial',
  targetPath: '/site/website/articles/2020/12/top-books-for-young-women-copy-2',
  eventType: 'CONTENT_EVENT'
}
```

Event 2:
```js
{
  timestamp: 1709153527627,
  user: { ... },
  siteId: 'editorial',
  targetPath: '/site/website/articles/2020/12/top-books-for-young-women-copy-2/index.xml',
  eventType: 'CONTENT_EVENT'
}
```

**Comment by rart** [comment]
@sumerjabri assigning back to you for re-triage.

**Comment by sumerjabri** [comment]
@jmendeza, please investigate to see if we can transmit a single event in the back and omit the folder event.

**Comment by jmendeza** [comment]
It would be complicated/dirty to omit the folder creation event, since we are actually performing the two operations separately (i.e.: when we are creating the folder we don't know if it is just the folder, or part of a larger content write).

While reproducing this issue, we observed that the race condition appears when there is enough time between the events for the UI to retrieve the children of the path before the write of the page.
When that happens, the UI will request the path children and get the folder.
On next event, after the page creation, the UI will request the children again and this time get a page item. However, it will fail to update the sidebar tree to reflect the change.

Moreover, consider the scenario where a folder is created and then a page is created for the same path in a different API call. The page creation will not be reflected in the sidebar until Studio is reloaded in the browser.

@rart @sumerjabri

**Comment by rart** [comment]
Note for testers: This PR changes a util used across the UI. Its impact spans all navigators, create folder dialog, quick create, the item selector that's in several places (e.g. opening the navigator from the preview address bar), and other various places.

**Comment by Euquimides** [comment]
Hi! Updated and rechecked this issue. Fix verified for Crafter v4.1.3. Thank you! I will keep a watch on the changes mentioned by @rart. Will update the comment once the fix is merged for v4.2.

**Comment by Euquimides** [comment]
Hi! Updated and rechecked this issue. Fix verified. Thank you! Closing this ticket.
